addons/point_of_sale/static/tests/unit/related_models_test.js

```diff
@@ -0,0 +1,948 @@
+            const c1 = models["product.category"].create({});
+            c1.update({ product_ids: [["link", p1, p2]] });
+            assert.ok(c1.product_ids.includes(p1), "Product should be linked to the category");
+            assert.ok(c1.product_ids.includes(p2), "Product should be linked to the category");
+            assert.equal(p1.category_id, c1, "Product should be updated with new category");
+        });
+
+        QUnit.test("update operation, unlink many2one", (assert) => {
+            const p1 = models["product.product"].create({ category_id: {} });
+            const c1 = p1.category_id;
+            assert.ok(c1, "Product should be linked to the category");
+            assert.deepEqual(c1.product_ids, [p1], "Category should be linked to the product");
+
+            p1.update({ category_id: undefined });
+            assert.equal(p1.category_id, undefined, "Product should be unlinked from the category");
+            assert.notOk(
+                c1.product_ids.includes(p1),
+                "Product should be unlinked from the category"
+            );
+        });
+
+        QUnit.test("update operation, unlink one2many", (assert) => {
+            const p1 = models["product.product"].create({});
+            const c1 = models["product.category"].create({});
+            c1.update({ product_ids: [["link", p1]] });
+            assert.ok(c1.product_ids.includes(p1), "Product should be linked to the category");
+            assert.equal(p1.category_id, c1, "Product should be linked to the category");
+
+            c1.update({ product_ids: [["unlink", p1]] });
+            assert.notOk(
+                c1.product_ids.includes(p1),
+                "Product should be unlinked from the category"
+            );
+            assert.equal(p1.category_id, undefined, "Product should be unlinked from the category");
+        });
+
+        QUnit.test("update operation, Clear one2many", (assert) => {
+            const category = models["product.category"].create({ name: "Category" });
+            models["product.product"].create({ name: "Product 1", category_id: category });
+            models["product.product"].create({ name: "Product 2", category_id: category });
+            models["product.category"].update(category, { product_ids: [["clear"]] });
+            const updatedCategory = models["product.category"].read(category.id);
+            assert.equal(
+                updatedCategory.product_ids.length,
+                0,
+                "All products should be unlinked from the category"
+            );
+        });
+
+        QUnit.test("update operation, Clear many2one", (assert) => {
+            const category = models["product.category"].create({ name: "Category" });
+            const product = models["product.product"].create({
+                name: "Product 1",
+                category_id: category,
+            });
+            models["product.product"].update(product, { category_id: [["clear"]] });
```

Comment:
```suggestion
            models["product.product"].update(product, { category_id: undefined });
```
only use the commands 'link', 'unlink', 'create' and 'clear' on x2many fields.